src/System.Memory/tests/Span/Overflow.cs

```diff
@@ -60,6 +60,208 @@ public static void IndexOverflow()
+        {
+            IntPtr huge;
+            try
+            {
+                huge = Marshal.AllocHGlobal(new IntPtr(ThreeGiB));
+            }
+            catch (Exception)
+            {
+                return;  // It's not implausible to believe that a 3gb allocation will fail - if so, skip this test to avoid unnecessary test flakiness.
+            }
+
+            try
+            {
+                unsafe
+                {
+                    int
+                        GuidThreeGiBLimit = (int)(ThreeGiB / sizeof(Guid)),
+                        GuidTwoGiBLimit = (int)(TwoGiB / sizeof(Guid)),
+                        GuidOneGiBLimit = (int)(OneGiB / sizeof(Guid));
+
+                    Span<Guid> mutable = new Span<Guid>((void*)huge, GuidThreeGiBLimit);
+                    ReadOnlySpan<Guid> span = new ReadOnlySpan<Guid>((void*)huge, GuidThreeGiBLimit);
+                    Guid guid = Guid.NewGuid();
+                    ReadOnlySpan<Guid> slice = span.Slice((GuidTwoGiBLimit + 1));
+                    mutable[GuidTwoGiBLimit + 1] = guid;
+                    Assert.Equal(guid, slice[0]);
+
+                    slice = span.Slice(GuidOneGiBLimit).Slice(1).Slice(GuidOneGiBLimit);
+                    Assert.Equal(guid, slice[0]);
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(huge);
+            }
+        }
+
+        [Fact]
+        public static void ReadOnlySliceStartLengthInt32Overflow()
+        {
+            IntPtr huge;
+            try
+            {
+                huge = Marshal.AllocHGlobal(new IntPtr(ThreeGiB));
+            }
+            catch (Exception)
+            {
+                return;  // It's not implausible to believe that a 3gb allocation will fail - if so, skip this test to avoid unnecessary test flakiness.
+            }
+
+            try
+            {
+                unsafe
+                {
+                    int
+                        GuidThreeGiBLimit = (int)(ThreeGiB / sizeof(Guid)),
+                        GuidTwoGiBLimit = (int)(TwoGiB / sizeof(Guid)),
+                        GuidOneGiBLimit = (int)(OneGiB / sizeof(Guid));
+
+                    Span<Guid> mutable = new Span<Guid>((void*)huge, GuidThreeGiBLimit);
+                    ReadOnlySpan<Guid> span = new ReadOnlySpan<Guid>((void*)huge, GuidThreeGiBLimit);
+                    Guid guid = Guid.NewGuid();
+                    ReadOnlySpan<Guid> slice = span.Slice(GuidTwoGiBLimit + 1, 20);
+                    mutable[GuidTwoGiBLimit + 1] = guid;
+                    Assert.Equal(guid, slice[0]);
+
+                    slice = span.Slice(GuidOneGiBLimit).Slice(1).Slice(GuidOneGiBLimit);
+                    Assert.Equal(guid, slice[0]);
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(huge);
+            }
+        }
+
+        [Fact]
+        public static void ReadOnlyArrayCtorStartLengthOverflow()
```

Comment:
I don't think this test belongs here. It isn't really an overflow, like the other tests.

It probably fits better with the AsSpan tests:
https://github.com/dotnet/corefx/blob/ec1671fd7feffe1070a94d9d86379e0dd95ddcdf/src/System.Memory/tests/Span/AsSpan.cs#L152

```C#
        [Fact]
        public void GuidArrayAsSpanWithStartAndLength()
        {
            var arr = new Guid[20];

            ReadOnlySpan<Guid> slice = arr.AsSpan().Slice(2, 2);
            Guid guid = Guid.NewGuid();
            slice[1] = guid;

            Assert.Equal(guid, arr[3]);
        }
```